kore/src/Kore/Repl.hs

```diff
@@ -96,26 +99,39 @@ runRepl
     -- ^ list of axioms to used in the proof
     -> [claim]
     -- ^ list of claims to be proven
-    -> InitialScript
-    -- ^ optional initial script
+    -> ReplScript
+    -- ^ optional script
+    -> ReplMode
+    -- ^ mode to run in
     -> Simplifier ()
 runRepl
     tools simplifier predicateSimplifier axiomToIdSimplifier
-    axioms' claims' initScript
+    axioms' claims' initScript replMode
   = do
-    let mscript = unInitialScript initScript
-    newState <- maybe (pure state) (parseEvalScript state) mscript
-    replGreeting
-    evalStateT (whileM repl0) newState
+    let mscript = unReplScript initScript
+    mNewState <-
+        maybe (pure . pure $ state) (parseEvalScript state) mscript
```

Comment:
This isn't just parsing the script, right? It also evaluates it, with the result being `mNewState`. So perhaps this should be refactored into a new function, `evaluateScript` like: `mNewState <- evaluateScript mscript` or something.